src/lib/server/xlat_eval.c

```diff
@@ -284,25 +284,15 @@ static xlat_action_t xlat_eval_one_letter(TALLOC_CTX *ctx, fr_cursor_t *out, REQ
 	 */
 
 	case 'c': /* Current epoch time seconds */
-	{
-		struct timeval now;
-
-		gettimeofday(&now, NULL);
-
 		MEM(value = fr_value_box_alloc(ctx, FR_TYPE_UINT64, NULL, false));
-		value->datum.uint64 = (uint64_t)now.tv_sec;
-	}
+		value->datum.uint64 = (uint64_t)fr_time_to_sec(fr_time());
+
 		break;
 
 	case 'C': /* Current epoch time microsecond component */
-	{
-		struct timeval now;
-
-		gettimeofday(&now, NULL);
-
 		MEM(value = fr_value_box_alloc(ctx, FR_TYPE_UINT32, NULL, false));
-		value->datum.uint32 = (uint64_t)now.tv_usec;
-	}
+		value->datum.uint32 = (uint64_t)fr_time_to_sec(fr_time());
```

Comment:
This is microseconds not seconds, should be fr_time_to_usec(), apologies if I made a typo in my last comment...

Reply:
value->datum.uint64 = (uint64_t)fr_time_to_sec(fr_time());